src/test/unit/encoders/ScalarTest.cpp

```diff
@@ -0,0 +1,225 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+ * with Numenta, Inc., for a separate license for this software code, the
+ * following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * ---------------------------------------------------------------------
+ */
+
+/** @file
+ * Unit tests for the ScalarEncoder and PeriodicScalarEncoder
+ */
+
+#include <string>
+#include <vector>
+#include <nupic/ntypes/Array.hpp>
+#include <nupic/encoders/Scalar.hpp>
+#include "gtest/gtest.h"
+
+using namespace nupic;
+
+template <typename T>
+std::string vec2str(std::vector<T> vec)
+{
+  std::ostringstream oss("");
+  for (size_t i = 0; i < vec.size(); i++)
+    oss << vec[i];
+  return oss.str();
+}
+
+std::vector<UInt> getEncoding(Encoder& e, Real64 input)
+{
+  Array inputArray = Array(NTA_BasicType_Real64);
+  inputArray.allocateBuffer(1);
+  ((Real64*) inputArray.getBuffer())[0] = input;
+
+  auto actualOutput = std::vector<UInt>(e.getWidth());
+  e.encodeIntoArray(inputArray, &actualOutput[0], false);
+  return actualOutput;
+}
+
+typedef struct _SCALAR_VALUE_CASE
```

Comment:
Good call. You can tell most of my C++ experience happened in 20-year-old codebases.